st2client/st2client/commands/trace.py

```diff
@@ -163,22 +164,31 @@ def run(self, args, **kwargs):
                 kwargs['sort_asc'] = True
             elif args.sort_order in ['desc', 'descending']:
                 kwargs['sort_desc'] = True
-
-        return self.manager.query(limit=args.last, **kwargs)
+        result, count = self.manager.query(limit=args.last, **kwargs)
+        return (result, count)
 
     def run_and_print(self, args, **kwargs):
-        instances = self.run(args, **kwargs)
+        instances, count = self.run(args, **kwargs)
         if instances and len(instances) == 1:
             # For a single Trace we must include the components unless
             # user has overriden the attributes to display
             if args.attr == self.display_attributes:
                 args.attr = ['all']
             self.print_trace_details(trace=instances[0], args=args)
         else:
-            self.print_output(reversed(instances), table.MultiColumnTable,
-                              attributes=args.attr, widths=args.width,
-                              json=args.json, yaml=args.yaml,
-                              attribute_transform_functions=self.attribute_transform_functions)
+            if args.json or args.yaml:
```

Comment:
It looks like this is repeated in quite some places so eventually it will need some refactor, but it should be fine for now.